packages/toast/src/toast.ts

```diff
@@ -0,0 +1,207 @@
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+    html,
+    CSSResultArray,
+    TemplateResult,
+    LitElement,
+    property,
+    PropertyValues,
+} from 'lit-element';
+import '@spectrum-web-components/button';
+import '@spectrum-web-components/icon';
+import '@spectrum-web-components/icons';
+
+import toastStyles from './toast.css.js';
+
+export const toastVariants = [
+    'negative',
+    'positive',
+    'info',
+    'error',
+    'warning',
+];
+
+/**
+ * @slot - The toast content
+ */
+
+export class Toast extends LitElement {
+    public static get styles(): CSSResultArray {
+        return [toastStyles];
+    }
+
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @property({ type: Boolean, reflect: true })
+    public disappearing = false;
+
+    /**
+     * The variant applies specific styling when set to `negative`, `positive`, `info`, `error`, or `warning`.
+     * `variant` attribute is removed when not matching one of the above.
+     *
+     * @param {String} variant
+     */
+    @property({ type: String })
+    public set variant(variant: string) {
+        if (variant === this.variant) {
+            return;
+        }
+        const oldValue = this.variant;
+        if (toastVariants.includes(variant)) {
+            this.setAttribute('variant', variant);
+            this._variant = variant;
+        } else {
+            this.removeAttribute('variant');
+            this._variant = '';
+        }
+        this.requestUpdate('variant', oldValue);
+    }
+
+    public get variant(): string {
+        return this._variant;
+    }
+
+    private _variant = '';
+
+    private renderIcon(variant: string): TemplateResult {
+        let label = '';
+        let name = '';
+        switch (variant) {
+            case 'info':
+                label = 'Information';
+                name = 'ui:InfoSmall';
+                break;
+            case 'negative':
+            case 'error': // deprecated
+            case 'warning': // deprecated
+                label = 'Negative';
+                name = 'ui:AlertSmall';
+                break;
+            case 'positive':
+            case 'success': // deprecated
+                label = 'Positive';
+                name = 'ui:SuccessSmall';
+                break;
+            default:
+                return html``;
+        }
+        return html`
+            <sp-icon
+                class="type"
+                label=${label}
+                name=${name}
+                size="s"
+            ></sp-icon>
+        `;
+    }
+
+    private countdownStart = 0;
+    private nextCount = -1;
+    private countdownLength = 6000;
```

Comment:
should we make this countdown length a property? seems like something that users of this component might want to control

Reply:
agreed, especially if for some reason you have long text. The guideline says "at least 5 seconds" so it might be good to enforce a minimum if so.